Update developer's guide (#354)
PEtab-dev/libpetab-python: doc/development.rst

--------------------

We follow `numpy's Python support policy <https://numpy.org/neps/nep-0029-deprecation_policy.html>`_.

Release process
---------------

1. Update the version number in ``petab/version.py``.

> 2. Update the changelog in ``doc/CHANGELOG.md``.

Reviewer: ```suggestion
2. Update the changelog in ``doc/CHANGELOG.md``. The update content can be generated automatically: draft a new dummy GitHub release with a dummy tag and the ``develop`` branch, then click "Generate release notes".
```

> 3. Create a pull request with the changes to the main branch.

Reviewer: Is this from `develop`? Or some `release_v101` like
```
git checkout develop
git pull
git checkout -b release_v101
git add petab/version.py doc/CHANGELOG.md
git commit -m 'bump version; update changelog'
```

If the latter then some more info is needed I think. "simple merge" to main, then merge main into develop.

Author: I'd leave that to the releasing person and assume some basic familiarity with git/github.

4. Once the pull request is merged, create a new release on GitHub.
   Make sure to set the tag to the version number prefixed with 'v'
   (e.g., ``v1.0.0``), and the release title to ``libpetab-python $RELEASE_TAG``
   (e.g., ``libpetab-python v1.0.0``).

> 5. The release will be automatically uploaded to PyPI through a GitHub actions
>    workflow.

Reviewer: ```suggestion
5. Check that the release is now available on PyPI. The upload to PyPI is performed automatically by a GitHub Actions workflow, which may take a few minutes to complete.
```

Style guide
-----------

Code style
~~~~~~~~~~

We use pre-commit with ruff to enforce code style. To install pre-commit and
the pre-commit hooks, run:

.. code-block:: bash

    pip install pre-commit
    pre-commit install

To run the pre-commit checks manually on all, not just the modified files, run:

.. code-block:: bash

    pre-commit run --all-files

Documentation style
~~~~~~~~~~~~~~~~~~~

We use `Sphinx <https://www.sphinx-doc.org/>`_ to generate the documentation.
The documentation is written in `reStructuredText <https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html>`_.

> We use the `sphinx docstring-style <https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html>`__ for new code.

Reviewer: ```suggestion
We use the `sphinx docstring-style <https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html>`__ for new code.
The ``:param [ParamName]:`` and ``:return:`` statements are important when applicable.
Manual type annotations (``:type  [ParamName]:``) are redundant and should be avoided.
```
